src/helpers/format/format.ts

```diff
@@ -0,0 +1,803 @@
+ */
+function removeTrailingZeroes(numberString: string): string | undefined {
+  let i = numberString.length - 1;
+  while (i >= 0 && numberString[i] === "0") {
+    i--;
+  }
+  return numberString.slice(0, i + 1) || undefined;
+}
+
+const leadingZeroesRegexp = /^0+/;
+
+/**
+ * Limit the size of the decimal part of a number to the given number of digits.
+ */
+function limitDecimalDigits(
+  decimalDigits: string,
+  maxDecimals: number
+): {
+  integerDigits: string;
+  decimalDigits: string | undefined;
+} {
+  let integerDigits = "0";
+  let resultDecimalDigits: string | undefined = decimalDigits;
+
+  // Note : we'd want to simply use number.toFixed() to handle the max digits & rounding,
+  // but it has very strange behaviour. Ex: 12.345.toFixed(2) => "12.35", but 1.345.toFixed(2) => "1.34"
+  let slicedDecimalDigits = decimalDigits.slice(0, maxDecimals);
+  const i = maxDecimals;
+
+  if (Number(decimalDigits[i]) < 5) {
+    return { integerDigits, decimalDigits: slicedDecimalDigits };
+  }
+
+  // round up
+  const leadingZeroes = slicedDecimalDigits.match(leadingZeroesRegexp)?.[0] || "";
+  const slicedRoundedUp = (Number(slicedDecimalDigits) + 1).toString();
+  const withoutLeadingZeroes = slicedDecimalDigits.slice(leadingZeroes.length);
+  // e.g. carry over from 99 to 100
+  const carryOver = slicedRoundedUp.length > withoutLeadingZeroes.length;
+  if (carryOver && !leadingZeroes) {
+    integerDigits = "1";
+    resultDecimalDigits = undefined;
+  } else if (carryOver) {
+    resultDecimalDigits = leadingZeroes.slice(0, -1) + slicedRoundedUp;
+  } else {
+    resultDecimalDigits = leadingZeroes + slicedRoundedUp;
+  }
+
+  return { integerDigits, decimalDigits: resultDecimalDigits };
+}
+
+/**
+ * Split numbers into decimal/integer digits using Intl.NumberFormat.
+ * Supports numbers with a lot of digits that are transformed to scientific notation by
+ * number.toString(), but is slow.
+ */
+function splitNumberIntl(
+  value: number,
+  maxDecimals: number = MAX_DECIMAL_PLACES
+): { integerDigits: string; decimalDigits: string | undefined } {
+  let formatter = numberRepresentation[maxDecimals];
+  if (!formatter) {
+    formatter = new Intl.NumberFormat("en-US", {
+      maximumFractionDigits: maxDecimals,
+      useGrouping: false,
+    });
+    numberRepresentation[maxDecimals] = formatter;
+  }
+
+  const [integerDigits, decimalDigits] = formatter.format(value).split(".");
+  return { integerDigits, decimalDigits };
+}
+
+/** Convert a number into a string, without scientific notation */
+export function numberToString(number: number, decimalSeparator: string): string {
+  const { integerDigits, decimalDigits } = splitNumber(number, 20);
+  return decimalDigits ? integerDigits + decimalSeparator + decimalDigits : integerDigits;
+}
+
+/**
+ * Check if the given format is a time, date or date time format. Only check the first part of a multi-part format.
+ */
+export const isDateTimeFormat = memoize(function (format: Format) {
```

Comment:
Just to have a proper name for the function in performance profiles. Otherwise it appears as `(anonymous)` and it's mixed together with all other anonymous functions
```suggestion
export const isDateTimeFormat = memoize(function isDateTimeFormat(format: Format) {
```

Reply:
sure 👍